sw/device/lib/dif/dif_kmac.c

```diff
@@ -290,6 +290,62 @@ dif_kmac_result_t dif_kmac_mode_sha3_start(dif_kmac_t *kmac,
   return kDifKmacOk;
 }
 
+dif_kmac_result_t dif_kmac_mode_shake_start(dif_kmac_t *kmac,
+                                            dif_kmac_mode_shake_t mode) {
+  if (kmac == NULL) {
+    return kDifKmacBadArg;
+  }
+
+  // Set key strength and calculate rate (r). The digest length (d) is set to 0
+  // to indicate that there is no fixed digest length.
+  uint32_t kstrength;
+  switch (mode) {
+    case kDifKmacModeShakeLen128:
+      kstrength = KMAC_CFG_KSTRENGTH_VALUE_L128;
+      kmac->offset = 0;
```

Comment:
I think the assignment to `offset` and `d` can be factiored out?

Reply:
Yeah, I was in two minds whether to do it but I think you're right. Done.